streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -598,69 +659,103 @@ public boolean test(final K1 key, final V1 value) {
         ValueJoiner<? super V, ? super V1, ? extends R> joiner,
         JoinWindows windows) {
 
-        return leftJoin(other, joiner, windows, null, null, null);
+        return leftJoin(other, withKey(joiner), windows, null, null, null);
+    }
+
+    @Override
+    public <VO, VR> KStream<K, VR> leftJoin(final KStream<K, VO> otherStream,
+                                            final ValueJoinerWithKey<? super K, ? super V, ? super VO, ? extends VR> joinerWithKey,
```

Comment:
i think we should just stick with `joiner` for the name of this param. here and elsewhere